pandas/indexes/base.py

```diff
@@ -340,6 +339,8 @@ def _shallow_copy(self, values=None, **kwargs):
         values : the values to create the new Index, optional
         kwargs : updates the default attributes for this Index
         """
+    @Appender(_index_shared_docs['_shallow_copy'])
+    def _shallow_copy(self, values=None, **kwargs):
```

Comment:
I don't know if we have _shallow_defined for any other classes if so need to add docs like u did here